### PR TITLE
feat: Profile UI Changes + Added Edit functionality

### PR DESCRIPTION
**Pull request by EiriniMits**
Changes: Profile UI Changes, Added an ImageButton for edit profile in the main profile page and deleted the item "Edit Profile" from the menu.

Screenshots for the change:
screenshot_1546555733
screenshot_1546555737

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt [hunk 1]
```diff
@@ -86,9 +87,15 @@ class ProfileFragment : Fragment() {
 
                 Picasso.get()
                         .load(it.avatarUrl)
-                        .placeholder(requireDrawable(requireContext(), R.drawable.ic_person_black_24dp))
+                        .placeholder(requireDrawable(requireContext(), R.drawable.ic_account_circle_grey_24dp))
                         .transform(CircleTransform())
                         .into(rootView.avatar)
+
+                val imageButton: ImageButton = view?.findViewById(R.id.edit_profile_button) as ImageButton
```
[the remark this answers is not in the document]
Comment (nikit19) [reply app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt]: No need of findViewById, directly reference the view through Synthetic binding

File: app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt [hunk 2]
```diff
@@ -86,9 +88,15 @@ class ProfileFragment : Fragment() {
 
                 Picasso.get()
                         .load(it.avatarUrl)
-                        .placeholder(requireDrawable(requireContext(), R.drawable.ic_person_black_24dp))
+                        .placeholder(requireDrawable(requireContext(), R.drawable.ic_account_circle_grey_24dp))
                         .transform(CircleTransform())
                         .into(rootView.avatar)
+
+                val imageButton: ImageButton = rootView.edit_profile_button
+                imageButton?.setOnClickListener {
```
Comment (nikit19) [inline app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt]: `rootView.edit_profile_button.setOnClickListener` should be written as

File: app/src/main/res/layout/fragment_profile.xml [hunk 3]
```diff
@@ -19,25 +19,39 @@
...
+                android:id="@+id/name"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:text="@string/user_name"
+                android:layout_centerInParent="true"
+                android:textColor="@color/black"
+                android:textSize="@dimen/text_size_expanded_title" />
+
+            <ImageButton
+                android:id="@+id/edit_profile_button"
```
Comment (nikit19) [inline app/src/main/res/layout/fragment_profile.xml]: should be in camelCase

File: app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt [hunk 4]
```diff
@@ -86,9 +88,14 @@ class ProfileFragment : Fragment() {
...
                         .load(it.avatarUrl)
-                        .placeholder(requireDrawable(requireContext(), R.drawable.ic_person_black_24dp))
+                        .placeholder(requireDrawable(requireContext(), R.drawable.ic_account_circle_grey_24dp))
                         .transform(CircleTransform())
                         .into(rootView.avatar)
+
+                rootView.editProfileButton.setOnClickListener {
+                    findNavController(rootView).navigate(R.id.editProfileFragment)
+                }
+
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt]: extra

File: app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt [hunk 5]
```diff
@@ -14,6 +14,7 @@ import android.view.ViewGroup
 import androidx.appcompat.app.AppCompatActivity
 import androidx.core.view.isVisible
 import androidx.fragment.app.Fragment
+import android.widget.ImageButton
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt]: extra

File: app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt (hunk 2 above)
Comment (EiriniMits) [reply app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt]: > `rootView.edit_profile_button.setOnClickListener` should be written as

@nikit19 You are right! I corrected it.

File: app/src/main/res/layout/fragment_profile.xml [hunk 6]
```diff
@@ -19,25 +19,39 @@
...
+            <ImageButton
+                android:id="@+id/editProfileButton"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:background="@null"
+                android:layout_toRightOf="@+id/name"
+                android:layout_marginLeft="@dimen/layout_margin_large"
+                android:layout_centerInParent="true"
+                app:srcCompat="@drawable/ic_create_black_24dp" />
+        </RelativeLayout>
```
Comment (iamareebjamal) [inline app/src/main/res/layout/fragment_profile.xml]: Replace with LinearLayout

Comment (EiriniMits) [reply app/src/main/res/layout/fragment_profile.xml]: > Replace with LinearLayout

Is it a good idea to use LinearLayout when you have a TextView and an ImageButton at the same row? I mean the layout in the row will be spoiled. Sorry if that's a silly question.

Comment (iamareebjamal) [reply app/src/main/res/layout/fragment_profile.xml]: It won't be spoiled. In fact, this layout is so simple that it doesn't even need to be a relative layout

Comment (EiriniMits) [reply app/src/main/res/layout/fragment_profile.xml]: > It won't be spoiled. In fact, this layout is so simple that it doesn't even need to be a relative layout

Thank you for the help! I made all the necessary corrections

**Comment by iamareebjamal** [comment]
@liveHarshit Review please

@EiriniMits There should be a space after comma in one of your changes. Please correct it

File: app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt [hunk 7]
```diff
@@ -86,9 +87,13 @@ class ProfileFragment : Fragment() {
 
                 Picasso.get()
                         .load(it.avatarUrl)
-                        .placeholder(requireDrawable(requireContext(), R.drawable.ic_person_black_24dp))
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt]: If ic_person_black_24dp is not used now, delete it. Same for the string.

**Pull request merged**